pkg/apiserver/registry/security/workloadidentity/storage/tokenrequest.go

```diff
@@ -59,9 +60,19 @@ func (r *TokenRequestREST) Destroy() {
 // - referenced context object
 // - gardener installation
 func (r *TokenRequestREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, _ *metav1.CreateOptions) (runtime.Object, error) {
-	if len(r.issuer) == 0 {
+	if r.tokenIssuer == nil {
 		return nil, errors.New("workload identity token issuer is not configured and tokens cannot be issued")
 	}
+
+	if len(r.tokenIssuer.Issuer()) == 0 {
+		return nil, errors.New("workload identity no value provided for the iss claim")
```

Comment:
Can you clarify this error message? 😕

Reply:
The `TokenIssuer` has been reworked so that this check is no longer required here.